### PR TITLE
Add initial challenge pages

### PR DESCRIPTION
**Pull request by teenoh**
The pages added include

- Course Select Page
- Mode Select Page
- Opponent Select Page
- Challenge Info Page

### CODE REVIEW

File: components/challenge/constants.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+export const CHALLENGE_PAGES_HEADERS = {
+    'challengeInfoLabel': 'Challenge Info',
+    'chooseOpponentLabel': 'Choose Opponent',
+    'courseSelectLabel': 'Select Course',
+    'modeSelectLabel': 'Select Mode',
+    'opponentSelectLabel': 'Choose Opponent',
+};
+
+export const CLASSNAMES = {
+
```
Comment (Way2nnadi) [inline components/challenge/constants.js]: nit: delete unused line

File: components/challenge/constants.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    modeCard: 'mode-card',
+    modeCardIcon: 'mode-card__icon',
+    mt1 : 'mt-1',
+    mt4: 'mt-4',    
+    onlineIcon: 'online-icon',
+    opponentCard: 'opponent-card',
+    opponentCardImg: 'opponent-card__img',
+    opponentCardTextContent: 'opponent-card__text-content',
+    textCenter: 'text-center',
+
```
Comment (Way2nnadi) [inline components/challenge/constants.js]: nit: unused line

File: components/challenge/constants.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    courseSelect: {
+        subjects: [
+            { id: 1, name: 'Mathematics' },
+            { id: 2, name: 'English' },
+            { id: 3, name: 'Biology' },
+            { id: 4, name: 'Physics' },
+            { id: 5, name: 'Chemistry' },
+            { id: 6, name: 'Government' },
+        ],
+
```
Comment (Way2nnadi) [inline components/challenge/constants.js]: here as well

File: pages/admin/challenge/index.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import { components } from '../../../components/challenge';
+
+const { CourseSelect } = components;
+
+const ChallengeHome = () => (
+    <Layout selectedMenuItem="challenge">
+        <CourseSelect next='/admin/challenge/select-mode' />
+    </Layout>
+);
+
```
Comment (Way2nnadi) [inline pages/admin/challenge/index.js:16]: strings in ` constants.js`

File: pages/admin/challenge/select-mode.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import Layout from '../../../components/layout';
+
+const { ModeSelect } = components;
+
+const SelectModePage = () => (
+    <Layout selectedMenuItem="challenge">
+        <ModeSelect next='/admin/challenge/choose-opponent' />
+    </Layout>
+);
+
```
Comment (Way2nnadi) [inline pages/admin/challenge/select-mode.js:17]: strings in ` constants.js`

**Review by Way2nnadi — APPROVED** [review]
Minor fixes, but code looks great to merge down

File: pages/admin/challenge/select-mode.js (hunk 5 above)
Comment (teenoh) [reply pages/admin/challenge/select-mode.js:17]: can't exactly use constants here, because this is the route page, not a component

**Review by tosinamuda — APPROVED** [review]

**Review by enyeInc — APPROVED** [review]

**Pull request merged**